### PR TITLE
fallback to english if no comments in current language fixes #421

### LINKED ISSUE #421: Continuation from #414: comment defaults
On initial page load the comment language does not default to English if no comments are present in the user's language

### PR DESCRIPTION
**Pull request by adsworth**

### CODE REVIEW

**Comment by BernhardPosselt** [comment]
Looks good otherwise, will test once map filter is in :D

**Comment by adsworth** [comment]
@BernhardPosselt does this look better. :)

File: nextcloudappstore/core/static/assets/js/ui/app-detail.js [hunk 1]
```diff
@@ -27,8 +27,16 @@
                         ratingTarget.appendChild(template);
                     });
                 } else {
-                    let templateNoComments = document.importNode(ratingTemplateNoComments.content, true);
-                    ratingTarget.appendChild(templateNoComments);
+                    let langCode = global.id('comment_display_language_code');
+                    let options = Array.from(langCode.options);
+                    let fallback = options.filter( function (o) { return o.value == fallbackLanguageCode; } );
```
Comment (BernhardPosselt) [inline nextcloudappstore/core/static/assets/js/ui/app-detail.js]: I'd combine this and use short function syntax (also ===), e.g. 

```js
let fallback = Array.from(langCode.options)
    .filter((option) => option.value === fallbackLanguageCode)
```

Comment (adsworth) [inline nextcloudappstore/core/static/assets/js/ui/app-detail.js]: Yes, that's a lot nicer. 

I'll get the hang of it eventually, promise. :)

File: nextcloudappstore/core/static/assets/js/ui/app-detail.js [hunk 2]
```diff
@@ -27,8 +27,16 @@
                         ratingTarget.appendChild(template);
                     });
                 } else {
-                    let templateNoComments = document.importNode(ratingTemplateNoComments.content, true);
-                    ratingTarget.appendChild(templateNoComments);
+                    let langCode = global.id('comment_display_language_code');
+                    let options = Array.from(langCode.options)
+                                    .filter( (o) => { return o.value === fallbackLanguageCode; } );
```
Comment (BernhardPosselt) [inline nextcloudappstore/core/static/assets/js/ui/app-detail.js]: you can completely drop the block here and replace it with an expression, the following are equivalent:

```js
(o) => { return o.value === fallbackLanguageCode; }
(o) => o.value === fallbackLanguageCode
```

Comment (BernhardPosselt) [inline nextcloudappstore/core/static/assets/js/ui/app-detail.js]: Same in java btw ;)

**Comment by BernhardPosselt** [comment]
Pushed the fixes to your branch

**Comment by BernhardPosselt** [comment]
Will test and merge later on btw :)

**Pull request merged**